sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample05_ReadingEvents.md

```diff
@@ -0,0 +1,544 @@
+}
+```
+
+## Read events from a partition with a maximum wait time
+
+When using `ReadEventsFromPartitionAsync`, it can sometimes be advantageous to ensure that the [Async Enumerable](https://docs.microsoft.com/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8) returns control to the application's code in the loop body periodically, whether an event was available or not.  This allows for the application to detect when events are no longer being published and is often used for emitting heartbeat data as a health check for consumers.
+
+This example illustrates waiting for a maximum of one second for an event to be read; if no event was available, the loop ticks with an empty `PartitionEvent`.  Once the example loop has ticked a total of 10 times, with or without an event available, it will exit.
+
+```C# Snippet:EventHubs_Sample05_ReadPartitionWaitTime
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+var consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+
+var consumer = new EventHubConsumerClient(
+    consumerGroup,
+    connectionString,
+    eventHubName);
+
+try
+{
+    using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+    cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+    string firstPartition = (await consumer.GetPartitionIdsAsync(cancellationSource.Token)).First();
+    EventPosition startingPosition = EventPosition.Earliest;
+
+    int loopTicks = 0;
+    int maximumTicks = 10;
+
+    var options = new ReadEventOptions
+    {
+       MaximumWaitTime = TimeSpan.FromSeconds(1)
+    };
+
+    await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(
+        firstPartition,
+        startingPosition,
+        options))
+    {
+        if (partitionEvent.Data != null)
+        {
+            string readFromPartition = partitionEvent.Partition.PartitionId;
+            byte[] eventBodyBytes = partitionEvent.Data.EventBody.ToArray();
+
+            Debug.WriteLine($"Read event of length { eventBodyBytes.Length } from { readFromPartition }");
+        }
+        else
+        {
+            Debug.WriteLine("Wait time elapsed; no event was available.");
+        }
+
+        loopTicks++;
+
+        if (loopTicks >= maximumTicks)
+        {
+            break;
+        }
+    }
+}
+catch (TaskCanceledException)
+{
+    // This is expected if the cancellation token is
+    // signaled.
+}
+finally
+{
+    await consumer.CloseAsync();
+}
+```
+
+## Read events from a partition, starting from a specific date and time
+
+When reading event from a partition, consumers can request to begin reading from the partition's event stream at a specific point in time.  This example illustrates reading events starting with an hour prior to the current time, and will do so for 30 seconds regardless of how many events have been read.
```

Comment:
```suggestion
When reading event from a partition, consumers can request to begin reading from the partition's event stream at a specific point in time.  This example illustrates reading as many events as possible for a duration of 30 seconds starting with an hour prior to the current time.
```

Reply:
I tweaked the verbiage a bit, but wanted to keep the lead focus on "we're starting with stuff that was an hour old."    Hopefully, it fixes the awkwardness of what I had without causing confusion between "we're starting with an hour ago" and "we're cancelling this puppy after 30 seconds."